internal/fwserver/server_planresourcechange.go

```diff
@@ -215,6 +216,15 @@ func (s *Server) PlanResourceChange(ctx context.Context, req *PlanResourceChange
 		resp.PlannedState.Raw = modifiedPlan
 	}
 
+	// Skip plan modification for automatic deferrals
+	// unless ProviderDeferredBehavior.EnablePlanModification is true
+	if s.deferred != nil && req.ResourceBehavior.ProviderDeferred.EnablePlanModification == false {
+		logging.FrameworkDebug(ctx, "Provider has deferred response configured, automatically returning deferred response.")
+		resp.Deferred = &resource.Deferred{
+			Reason: resource.DeferredReason(s.deferred.Reason),
+		}
+	}
```

Comment:
Same thoughts about returning early from the `ReadResource` comment.

This is the right idea, we'll just need to run the framework-defined logic only here and then immediately return. There's two ways to approach it:
1. How you did it here, where we would need to duplicate the "marking all computed nil as unknown" logic call up here, then immediately return. If `EnablePlanModification == true`, then it'll be set by your second `if s.deferred != nil` below. This matches how [SDKv2 does it](https://github.com/hashicorp/terraform-plugin-sdk/blob/e90e8a2497cb1bced1d5614e65c904453244d18c/helper/schema/grpc_provider.go#L777-L792)
2. Add if statements to all provider-defined logic calls, which could get cumbersome and we'd need to remember to keep propogating that if we ever add more provider-defined logic to plan modification.

I'm currently leaning towards option 1

Brian did mention that we could potentially run the schema-based plan modification, which may make option 2 look more appealing.